### PR TITLE
hr_attendance_calendar fixes

### PR DESCRIPTION
**Pull request by samuel-fringeli**

### CODE REVIEW

File: hr_attendance_calendar/models/hr_attendance_day.py [hunk 1]
```diff
@@ -181,7 +184,26 @@ def _compute_break_total(self):
     @api.depends('worked_hours', 'due_hours')
     def _compute_extra_hours(self):
         for att_day in self.filtered('worked_hours'):
-            att_day.extra_hours = att_day.worked_hours - att_day.due_hours
+            extra_hours = att_day.worked_hours - att_day.due_hours
+            max_extra_hours = float(self.env['ir.config_parameter'].get_param(
+                'hr_attendance_calendar.max_extra_hours'))
+
+            if extra_hours > max_extra_hours:
```
Comment (eicher31) [inline hr_attendance_calendar/models/hr_attendance_day.py]: The max_extra_hours is for the sum of all extra hours, this comparison should be done in the employee.

File: hr_attendance_calendar/models/hr_attendance_day.py [hunk 2]
```diff
@@ -181,7 +184,26 @@ def _compute_break_total(self):
...
+                att_day.extra_hours = max_extra_hours
+            else:
+                att_day.extra_hours = extra_hours
```
Comment (eicher31) [inline hr_attendance_calendar/models/hr_attendance_day.py]: extra_hours should not be altered even if the limit had exceeded

File: hr_attendance_calendar/models/hr_attendance_day.py [hunk 3]
```diff
@@ -181,7 +184,26 @@ def _compute_break_total(self):
...
+
+    @api.multi
+    @api.depends('extra_hours')
+    def _compute_extra_hours_lost(self):
+        for att_day in self.filtered('worked_hours'):
+            extra_hours = att_day.worked_hours - att_day.due_hours
```
Comment (eicher31) [inline hr_attendance_calendar/models/hr_attendance_day.py]: already compute in att_day.extra_hours

**Review by eicher31 — CHANGES_REQUESTED** [review]
Do not hesitate to ask me if you don't understand some of my requests.

File: hr_attendance_calendar/models/hr_attendance_day.py (hunk 1 above)
Comment (samuel-fringeli) [reply hr_attendance_calendar/models/hr_attendance_day.py]: No, it's the setting we can configure in attendances settings (hr_attendance_calendar.max_extra_hours)

File: hr_attendance_calendar/models/hr_attendance_day.py (hunk 2 above)
Comment (samuel-fringeli) [reply hr_attendance_calendar/models/hr_attendance_day.py]: extra_hours is altered, but the difference goes in extra_hours_lost

File: hr_attendance_calendar/models/hr_attendance_day.py (hunk 3 above)
Comment (samuel-fringeli) [reply hr_attendance_calendar/models/hr_attendance_day.py]: yes, but extra_hours can already have the value of max_extra_hours at this state of the execution, so it's necessary to recompute this value

**Comment by ecino** [comment]
Conflicts with devel, the branch should be rebased.

**Review by ecino — APPROVED** [review]

**Pull request merged**